sky/serve/core.py

```diff
@@ -572,6 +574,83 @@ def status(
+                                           cluster_status=controller_status)
+
+    ssh_credentials = backend_utils.ssh_credential_from_yaml(
+        controller_handle.cluster_yaml, controller_handle.docker_user)
+    runner = command_runner.SSHCommandRunner(
+        controller_handle.head_ip,
+        port=controller_handle.head_ssh_port,
+        **ssh_credentials,
+    )
+    sky_logs_directory = os.path.expanduser(constants.SKY_LOGS_DIRECTORY)
+    run_timestamp = backend_utils.get_run_timestamp()
+
+    prepare_code = serve_utils.ServeCodeGen.prepare_replica_logs_for_download(
+        service_name, run_timestamp)
+    backend = backend_utils.get_backend_from_handle(controller_handle)
+    assert isinstance(backend, backends.CloudVmRayBackend)
+    assert isinstance(controller_handle, backends.CloudVmRayResourceHandle)
+    sky_logging.print('Preparing replica logs for download on controller')
+    prepare_returncode = backend.run_on_head(controller_handle,
+                                             prepare_code,
+                                             require_outputs=False,
+                                             stream_logs=False)
+    subprocess_utils.handle_returncode(
+        prepare_returncode, prepare_code,
+        'Failed to prepare replica logs to sync down.')
+    remote_service_dir_name = serve_utils.generate_remote_service_dir_name(
+        service_name)
+    dir_for_download = os.path.join(remote_service_dir_name, run_timestamp)
+    sky_logging.print('Downloading the replica logs')
+    runner.rsync(source=dir_for_download,
+                 target=sky_logs_directory,
+                 up=False,
+                 stream_logs=False)
+    remove_code = serve_utils.ServeCodeGen.remove_replica_logs_for_download(
+        service_name, run_timestamp)
+    remove_returncode = backend.run_on_head(controller_handle,
+                                            remove_code,
+                                            require_outputs=False,
+                                            stream_logs=False)
+    subprocess_utils.handle_returncode(
+        remove_returncode, remove_code,
+        'Failed to remove the replica logs for download on the controller.')
+
+    # We download the replica logs first because that download creates the
+    # timestamp directory, and we can just put the controller and load balancer
+    # logs in there. Otherwise, we would create the timestamp directory with
+    # controller and load balancer logs first, and then the replica logs
+    # download would overwrite it
+    controller_log_file_name = (
+        serve_utils.generate_remote_controller_log_file_name(service_name))
+    target_directory = os.path.join(sky_logs_directory, run_timestamp)
+    sky_logging.print('Downloading the controller logs')
+    runner.rsync(source=controller_log_file_name,
+                 target=target_directory,
+                 up=False,
+                 stream_logs=False)
```

Comment:
how about defining a function `sync_down_serve_logs` in the backend and call it here? feeling like we should not include those complex logic into `sky::serve::core` module. reference: https://github.com/skypilot-org/skypilot/blob/4436aed82b92119257c5cdc271ed45e1a867fe8b/sky/core.py#L667-L705